sw/device/examples/bridge/main.c

```diff
@@ -180,9 +184,25 @@ static void configure_spi_device(dif_spi_device_handle_t *spi_device) {
   CHECK_DIF_OK(dif_spi_device_set_flash_id(spi_device, flash_id));
 }
 
-static void event_loop(dif_spi_device_handle_t *spi_device) {
+static void event_loop(dif_spi_device_handle_t *spi_device,
+                       screen_context_t *screen_ctx) {
+  enum { kTimeout = 800 * 1000 };
+  ibex_timeout_t deadline = ibex_timeout_init(kTimeout);
+  CHECK_DIF_OK(dif_spi_device_clear_flash_busy_bit(spi_device));
   while (true) {
-    CHECK_DIF_OK(dif_spi_device_clear_flash_busy_bit(spi_device));
+    if (ibex_timeout_check(&deadline)) {
+      deadline = ibex_timeout_init(kTimeout);
+      screen_keep_alive(screen_ctx);
+    }
+
+    bool upload_pending;
+    CHECK_DIF_OK(dif_spi_device_irq_is_pending(
```

Comment:
Does the IRQ need to be cleared by ibex or does it go away when the upload is dealt with?